tests/test_client.py

```diff
@@ -423,3 +425,68 @@ def test_search_max_items_unlimited_default(self, api: Client) -> None:
         )
         items = list(search.items())
         assert len(items) > 100
+
+
+class MySign:
+    def __init__(self) -> None:
+        self.call_count = 0
+
+    def __call__(self, x: Any) -> None:
+        self.call_count += 1
+
+
+class TestSigning:
+    @pytest.mark.vcr  # type: ignore[misc]
+    def test_signing(self) -> None:
+        sign = MySign()
+        # sign is callable, but mypy keeps trying to interpret it as a "MySign" object.
+        client = Client.open(STAC_URLS["PLANETARY-COMPUTER"], modifier=sign)
+        assert client.modifier is sign  # type: ignore
```

Comment:
```suggestion
    def __call__(self, x: Modifiable) -> None:
        self.call_count += 1


class TestSigning:
    @pytest.mark.vcr  # type: ignore[misc]
    def test_signing(self) -> None:
        sign = MySign()
        # sign is callable, but mypy keeps trying to interpret it as a "MySign" object.
        client = Client.open(STAC_URLS["PLANETARY-COMPUTER"], modifier=sign)
        assert client.modifier is sign
```

The `mypy` error wasn't very helpful, but I was able to avoid the `# type: ignore` by modifying the `__call__` typing.

I think the type issue in line 449 below is actually stemming from the fact that `Client.get_collection` is typed. Fixing that seems out of scope for this PR, so I think we should leave that `# type: ignore` comment in place and see if we can come up with a more thorough fix later.